### DISCUSSION TITLE
Capacitor seems like to not install Userbase Cordova Plugin

### DISCUSSION THREAD

**Discussion by julien040** [ionic-team/capacitor#4425] [General] [answered]
Hi, I would like to use a Cordova plugin in my capacitor project.
I have run all commands asked I think.
But capacitor-runtime.js generate an error that it cannot found the required js file of the plugin.

The cordova plugin :
https://github.com/smallbets/userbase/tree/master/src/cordova-plugin-userbase

The error :
```js
E/Capacitor/Console: File: capacitor-runtime.js - Line 361 - Msg: module cordova-plugin-userbase.userbase-js not found

E/Capacitor: JavaScript Error: {"type":"js.error","error":{"message":"Uncaught module cordova-plugin-userbase.userbase-js not found","url":"capacitor-runtime.js","line":1155,"col":17,"errorObject":"\"module cordova-plugin-userbase.userbase-js not found\""}}

E/Capacitor/Console: File: capacitor-runtime.js - Line 1155 - Msg: Uncaught module cordova-plugin-userbase.userbase-js not found
```

Doctor :
```
npx cap doctor
   Capacitor Doctor

Latest Dependencies:

  @capacitor/cli: 2.4.7
  @capacitor/core: 2.4.7
  @capacitor/android: 2.4.7
  @capacitor/electron: 2.4.7
  @capacitor/ios: 2.4.7

Installed Dependencies:

  @capacitor/ios not installed
  @capacitor/cli 2.4.7
  @capacitor/android 2.4.7
  @capacitor/core 2.4.7
  @capacitor/electron not installed

[success] Android looking great! 👌
```

Commands I have run to (try to) install it :
```
npm install cordova-plugin-userbase --save

+ cordova-plugin-userbase@1.0.1
added 14 packages from 14 contributors and audited 74 packages in 7.379s


found 0 vulnerabilities


C:\Users\etc>npx cap sync
√ Copying web assets from www to android\app\src\main\assets\public in 59.93ms
√ Copying native bridge in 1.83ms
√ Copying capacitor.config.json in 1.73ms
  Found 1 Cordova plugin for android
    cordova-plugin-userbase (1.0.1)
√ copy in 148.96ms
√ Updating Android plugins in 8.78ms
  Found 1 Capacitor plugin for android:
    @capacitor-community/barcode-scanner (1.2.1)
  Found 1 Cordova plugin for android
    cordova-plugin-userbase (1.0.1)
√ update android in 55.52ms
√ Copying capacitor.js to web dir in 1.91ms
√ copy in 3.05ms
√ update web in 6.00μp
Sync finished in 0.223s

C:\Users\etc>npx cap copy android
√ Copying web assets from www to android\app\src\main\assets\public in 58.85ms
√ Copying native bridge in 1.59ms
√ Copying capacitor.config.json in 1.68ms
  Found 1 Cordova plugin for android
    cordova-plugin-userbase (1.0.1)
√ copy in 143.81ms
```

Thanks to anyone who see this. Hope some can help me.
Have a great day

**Comment by jcesarmobile** [comment, marked as answer]
This is a Capacitor bug.
To speed things up, Capacitor injects Cordova plugin files, but only do it for files in "plugins" folder.
userbase plugin copy a file to a `node_modules` folder, so it doesn't get injected, and the other plugin injected file depends on this one, since it's not injected it doesn't find the module.

I don't thin we will fix it as it would require injecting the node_modules folder and people might have other files there that are not part of Cordova plugins.

Try reporting an issue on their repository and ask if they can distribute this file from www folder instead of from node_modules
https://github.com/smallbets/userbase/blob/master/src/cordova-plugin-userbase/plugin.xml#L19

**Reply by julien040** [reply]
Thank you. You took time just for answering me and you are incredible for that.
Do you think if I manually copy the file, it will work ?

**Reply by jcesarmobile** [reply]
You’ll need to manually copy the file and update the plugin.xml to use the new location
